core/controllers/acl_decorators.py

```diff
@@ -1020,6 +1020,57 @@ def test_can_voiceover(self, exploration_id, **kwargs):
     return test_can_voiceover
 
 
+def can_save_exploration(handler):
+    """Decorator to check whether user can save exploration.
+
+    Args:
+        handler: function. The function to be decorated.
+
+    Returns:
+        function. The newly decorated function that checks if
+            a user has permission to save a given
+            exploration.
+    """
+
+    def test_can_save(self, exploration_id, **kwargs):
+        """Checks if the user can save the exploration.
+
+        Args:
+            exploration_id: str. The exploration id.
+            **kwargs: dict(str: *). Keyword arguments.
+
+        Returns:
+            *. The return value of the decorated function.
+
+        Raises:
+            NotLoggedInException: The user is not logged in.
+            PageNotFoundException: The page is not found.
+            UnauthorizedUserException: The user does not have
+                credentials to voiceover an exploration.
+        """
+
+        if not self.user_id:
+            raise base.UserFacingExceptions.NotLoggedInException
+
+        exploration_rights = rights_manager.get_exploration_rights(
+            exploration_id, strict=False)
+        if exploration_rights is None:
+            raise base.UserFacingExceptions.PageNotFoundException
+        can_voiceover = rights_manager.check_can_voiceover_activity(
```

Comment:
I think it might be cleaner to condense both of these into rights_manager.check_can_save_activity().